parser.go

```diff
@@ -532,7 +532,7 @@ func (p *Parser) parseOption(s *parseState, name string, option *Option, canarg
 		} else {
 			arg = s.pop()
 
-			if argumentIsOption(arg) && !(option.isSignedNumber() && len(arg) > 1 && arg[0] == '-' && arg[1] >= '0' && arg[1] <= '9') {
+			if argumentIsOption(arg) && !option.isValidValue(arg) {
```

Comment:
I'd expect the validator to run also if the argument isn't an option. So `argumentIsOption` would need to fold into `isValidValue` default case I think.